src/Billing/AutoBilling.php

```diff
@@ -0,0 +1,327 @@
+        } else {
+            $this->genMultiCode();
+        }
+    }
+
+    public function generateBilling($dx, $cpt)
+    {
+        $event_date = date("Y-m-d H:i:s");
+
+        /**
+         *  Aggregate information to fill billing table
+         */
+        $billingData = explode("-", $cpt);  //separate up billing information into code and text
+        $code = trim($billingData[0]);  // code that is in the calendar description with the decimalOpen
+        $text = $billingData[1]; //get the billing description from the code table
+
+
+        $icd10code = explode(" - ", $dx);
+        $icd10 = $icd10code[0];
+        $desc = $icd10code[1];
+
+        /**
+         * Grab Fee
+         * @returns int
+        */
+        $fees = $this->getCodeFee($code);
+
+        /**
+         * Grab the encounter to post charges using the event date
+         * This file has been tailored to other areas submitting data
+         * to create billing. I left them in case they are needed in the future
+         * like doing billing from the calendar
+         * @returns int
+         */
+       //$enc = $this->getEventEncounter($pid, $event_date);
+        $enc = $_SESSION['encounter'];
+        /**
+         * Grab the provider for the encounter from the event table
+         */
+        //$provider = $this->getEventProvider($pid, $event_date);
+        $provider = $_SESSION['authUserID'];
+        /**
+         * find out if there are any entries in the billing table for event date for this patient
+         * moved to calendar inc
+         *
+         */
+        $hasBilling = $this->findEventBilling($this->pid, $enc);
+
+        /**
+         * Gets the modifier assigned to the
+         * If there are modifiers in the code table
+         */
+        $modifier = $this->getModifier($code);
+        $modifier = str_replace(",", "", $modifier);
+
+        //insert billing if there is note and entry for this particular encounter
+        if (empty($hasBilling['encounter'])) {
+            if (!empty($code)) {
+                    $codetype = "CPT4";
+                    $this->insertCPTBilling($event_date, $code, $this->pid, $provider, $this->userid, $enc, $text, $fees, $icd10, $codetype, $modifier);
+            }
+
+            //Enter ICD10 if it exist
+            if (!empty($desc) && empty($res)) {
+                    $codetype = "ICD10";
+                    $this->insertICDBilling($event_date, $codetype, $icd10, $this->pid, $provider, $this->userid, $enc, $desc);
+            }
+        }
+    }// End of billing entries
+
+    /**
+     * @param $event_date
+     * @param $code
+     * @param $pid
+     * @param $provider
+     * @param $userid
+     * @param $enc
+     * @param $text
+     * @param $fees
+     * @param $icd
+     * @param $codetype
+     * @param $modifier
+     */
+    private function insertCPTBilling($event_date, $code, $pid, $provider, $userid, $enc, $text, $fees, $icd10, $codetype, $modifier)
```

Comment:
Future developers will thank you if you avoid more than 3-4 parameters for an function. Specially in this case you clearly are providing database record properties and the function is executing an record method. So why not have the caller give you table name and column name/value pairs? Then you could even make these inserts generic.

Reply:
@mdsupport as a junior developer, I am still learning the ways to do things. I was focused on getting the feature to work. There are a few columns that are not needed for every entry. I could void those. I will study the other code in the system and figure out how to do what you suggested. I see @adunsulag do that all the time. I will learn.

Reply:
@juggernautsei As stated, suggestion will benefit future developers as there is nothing wrong with your code.  Suggestion is to increase flexibility by passing e.g. a single assoc array (col name/value pairs) rather than just dropping parameters and hard coding their defaults - that would make the function even less usable in future. Benefit for the caller is if they have a blank or existing record, their code just fills/changes needed fields and call storage method (your function).

On the other hand if you anticipate this is highly specific situation, all the overhead is wasted effort.

Best.

Reply:
@juggernautsei Maybe you copied some of the code from existing codebase. In that case reviewers will need to decide if fields that already have default values still need to be initialized to the same (default) values in the code.

Reply:
Its a private method inside the class, not sure it really matters too much.  For better programming, I think the suggestion is good.  I personally prefer to keep my parameters at no more than 4-5 at most.  I prefer strongly typed objects to associative arrays but I've stuck with the associative array pattern in most places because we use it in the codebase everywhere.  It does increase the burden of documentation (where if there is no documentation of the parameters you have to violate information encapsulation by just diving into the method anyways).